lib/tom_queue/deferred_work_manager.rb

```diff
@@ -113,6 +121,24 @@ def stop
       deferred_set && deferred_set.interrupt
     end
 
-  end
+    # Register handlers to shut down the manager gracefully
+    def register_signal_handlers
+      Thread.main[:signals] = []
+
+      [:QUIT, :INT, :TERM].each do |sig|
+        trap(sig) do
+          Thread.main[:signals] << sig
```

Comment:
@jaxi I just got an email about a comment, but can't find it now. @cfabianski wrote the trap before the variable was declared, hence the error, but I don't see why something like this isn't an option:

    deferred_manager = TomQueue::DeferredWorkManager.new
    trap("SIGINT") { deferred_manager.stop }
    trap("SIGTERM") { deferred_manager.stop }
    deferred_manager.start

?